yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,203 @@
+
+func MonitorTargetYBHealth(yb TargetDBForMonitorHealth, skipDiskUsageHealthChecks utils.BoolStr, skipReplicationChecks utils.BoolStr, skipNodeHealthChecks utils.BoolStr, displayAlertFunc func(alertMsg string)) error {
+	nodeStatuses = make(map[string]bool)
+
+	var err error
+	loadBalancerEnabled, servers, err := yb.GetYBServers()
+	if err != nil {
+		return fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	for _, server := range servers {
+		nodeStatuses[server.Host] = true
+	}
+
+	var nodeAlert string
+	for {
+		nodeAlert, err = monitorNodesStatusAndAdapt(yb, loadBalancerEnabled, skipNodeHealthChecks)
+		if err != nil {
+			log.Errorf("error monitoring the node status and adapt: %v", err)
+		}
+
+		//skipping the disk usage monitoring for now, util the metrics function changes are made
+		err = monitorDiskUsageAndAbort(yb, true)
+		if err != nil {
+			log.Errorf("error monitoring the disk and memory: %v", err)
+		}
+
+		err = monitorReplicationOnTarget(yb, skipReplicationChecks)
+		if err != nil {
+			log.Errorf("error monitoring the replication: %v", err)
+		}
+
+		if nodeAlert != "" {
+			alertMsg := color.RedString(fmt.Sprintf("Alert!\n%s\n", nodeAlert))
+			displayAlertFunc(alertMsg)
+		}
+		time.Sleep(time.Duration(MONITOR_HEALTH_FREQUENCY_SECONDS) * time.Second)
+	}
+}
+
+/*
+1. if node goes down
+remove its connections from pool and if any current importBatch is running it will fail and automatic retry will take care of using the rest conns
+
+2. if node come back
+the new connections will automatically go to that node -- handled via conn-pool
+
+3. load balancer on the cluster
+No node status checks will happen as we put the load on the load balancer IP and it will take care of this adaptation automatically
+*/
+func monitorNodesStatusAndAdapt(yb TargetDBForMonitorHealth, loadBalancerEnabled bool, skip utils.BoolStr) (string, error) {
+	if bool(skip) || loadBalancerEnabled {
+		return "", nil
+	}
+	_, currentNodes, err := yb.GetYBServers()
+	if err != nil {
+		return "", fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	downNodes := make([]string, 0)
+
+	for node, up := range nodeStatuses {
+		if up && !lo.ContainsBy(currentNodes, func(t *tgtdb.TargetConf) bool {
+			return t.Host == node
+		}) {
+			downNodes = append(downNodes, node)
+		}
+	}
+	for _, conf := range currentNodes {
+		//Add the node in the main nodeStatuses initialised in the starting
+		//which weren't present the starting and during the run it came back.
+		if !slices.Contains(lo.Keys(nodeStatuses), conf.Host) {
+			nodeStatuses[conf.Host] = true
+		}
+	}
+	if len(downNodes) > 0 {
+		downNodeMsg := fmt.Sprintf("Following nodes are not healthy, please check them - [%v]", strings.Join(downNodes, ", "))
+		err := yb.RemoveConnectionsForHosts(downNodes)
+		if err != nil {
+			return downNodeMsg, fmt.Errorf("error while removing the connections for the down nodes[%v]: %v", downNodes, err)
+		}
+		return downNodeMsg, nil
```

Comment:
It might be worth passing the displayAlertFunc into these inner functions as well, instead of returning each msg, and then calling it in the main function. 

Reasons:
- You might want to send multiple alerts 
- Multiple functions (monitor*) might want to send alerts.